### DISCUSSION TITLE
Using SDK to get file with known path

### DISCUSSION THREAD

**Discussion by leetheman** [microsoftgraph/msgraph-sdk-dotnet#2215] [Q&A]
Hi all, I'm really struggling here with the C# SDK.  

Quite simply, I have a file path  for an item I have uploaded in SharePoint e.g

https://xxxx.sharepoint.com/sites/zzzzzzz/Shared%20Documents/file.xlsb

Is there a way to use the C# SDK to retreive this file?

So far I've tried using:

  var result = await graphClient.Drives[DRIVE_ID].Root.ItemWithPath(filePath).GetAsync();

but using the file path above, I get a "resource not found" error.  Any ideas?

**Comment by leetheman** [comment]
Just to add to this.

If I use the Graph Api endpoint:

https://graph.microsoft.com/v1.0/sites/{site-id}/drive/root/children  

I can get all the files and their Ids easily.

I would assume that this maps to:

graphClient.Sites[{site-id}].Drive.Root.Children

HOWEVER, Drive does not have a Root item unless I GetAsync.  Problem is, when I do, Drive.Root is null?

**Reply by andrueastman** [reply]
Thanks for raising this @leetheman 

Any chance the guidance at the link below is helpful? 
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md#drive-item-paths

Also, note that `drive/root/children` is actually a shorthand for `drive/items/root/children` so the `root` can be replaced with `items/root`
